src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,85 @@
-import React from 'react';
-import { Loader } from '../Loader';
+import { FC, useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Loader } from '..';
+import { useAppDispatch, useAppSelector } from '../../app/hook';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+export const TodoModal: FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  const dispatch = useAppDispatch();
+  const todo = useAppSelector(state => state.currentTodo);
+
+  const handleCloseModal = () => {
+    dispatch(currentTodoSlice.actions.setCurrentTodo(null));
+  };
+
+  useEffect(() => {
+    if (todo) {
+      setLoading(true);
+
+      getUser(todo.userId)
+        .then(setUser)
+        .finally(() => setLoading(false));
```

Comment:
Consider adding error handling for the `getUser` promise to manage potential API errors gracefully. This will improve the user experience by providing feedback in case of a failure.